### ISSUE TITLE
Composable menu - ensure examples/demo are appending menu correctly

### ISSUE THREAD

**Issue by nicolethoen** [patternfly/patternfly-react#7291]
Most of our composable menu demos are appending the menu to the document body.

The MenuToggle is a button by default - so we cannot append the Menu to the button or the Menu will be an interactable element inside an interactable button. 

Some possible suggestions:
- Make the menu toggle a div rather than a button for the sake of being able to append a menu to it
- build a container to place a menutoggle and menu inside
- reevaluate the appendto parent recommendation
